faycsic/main/hash.c

```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include "mbedtls/sha256.h"
+
+
+
+// Convert a byte array to a hex string for readability (optional)
+void to_hex(unsigned char *hash, char output[]) {
+    for (int i = 0; i < 32; i++) {
+        sprintf(output + (i * 2), "%02x", hash[i]);
+    }
+}
+
+// Perform double SHA-256 hashing
+void double_sha256(unsigned char *input, size_t len, uint8_t output[32]) {
+    unsigned char temp[32];
+
+    mbedtls_sha256(input, strlen((const char *)input), temp,0);
+    mbedtls_sha256(temp, strlen((const char *)temp), output,0);
+}
+
+int hash(unsigned char target[32], uint8_t block_header[80]) {
+    char target_output[32 * 2 + 1];
+    to_hex(target, target_output);
+    printf("Target:      %s\n\n", target_output);
+
+    unsigned char hash[32];
+    char hex_output[32 * 2 + 1];
+    for (int i = 0; i < 10; i++) {
+        block_header[0] = i;
+        double_sha256(block_header, sizeof(block_header), hash);
```

Comment:
Pretty sure the warning being thrown here has to do with:
* `sizeof(block_header)` is going to return what the size of `uint8_t` / `unsigned char *` - which is 1 byte
* `double_sha`'s second arg is the a `size_t len` 

If you're trying to pass in the length of `block_header`, you actually want it to be **_80_** bytes. So you can be fancy and do `sizeof(block_header) * 80` or just say 80...because at the end of the day - you want it to be 80 bytes.

Reply:
Even more - don't even bother passing in the length - we know we're only ever hashing block headers which are always 80 bytes